Feature/solver manager

Carmona-Anthony opened the pull request:
Ajout d'un manager pour la gestion des clusters générées ainsi que des détecteurs

Carmona-Anthony committed:
Ajout de minTime dans ConvoyDetector

Carmona-Anthony committed:
Ajout d'un manager pour la generation de clusters et des detecteurs

Carmona-Anthony committed:
Ajout des commentaires sur le solverManager

stardisblue committed:
change l'affichage des classes

stardisblue committed:
mise à jour du fonctionnement interne de debug

stardisblue committed:
optimise ConvoyDetector et affichage de Convoy

stardisblue committed:
renomme SolverManager en Solver, optimise clusterGenerator et ajoute ArrayUtils

stardisblue commented:
dans `Solver`, je comprends pas la logique de `generateCluster() `qui retourne `generatedClusters`, mais celui-ci n'est pas utilisé 😢

stardisblue committed:
renomme le nom du package de motifs à pattern

Carmona-Anthony committed:
Change Convoy to Set of transactions

Carmona-Anthony committed:
Change commentaries on convoy class

Carmona-Anthony committed:
Debug Convoy detection

stardisblue committed:
maj convoycluster

Carmona-Anthony committed:
Ajout de la classe ClosedSwarm Detector et de son fonctionnement

Carmona-Anthony committed:
Ajout de la classe ClosedSwarmDetector

stardisblue committed:
corrige `ClusterGenerator` :

- Database d'origine renommée en `defaultDatabase` et bougé dans le constructeur
- implemente `OccurenceDeriver` en tant que database qui est passée en paramètre dans les fonctions de `ClusterGenerator`
- ajout de fonctions clean dans `Database`, `Cluster` et `Transactions` qui supprime toutes les liaisons entre les objets.
- dépreciation de plusieurs fonctions

stardisblue committed:
ajoute des fonctions dans debug :
- `println(str, obj)`  affiche le nom et ensuite l'objet
- `printTitle(str)` affiche str en tant que titre

stardisblue committed:
ajoute `PatternGenerator` (non stable)

stardisblue committed:
renommage et documentation

stardisblue committed:
late night update :
- supprime les variable inutiles dans updateClusters
- maj de l'indentation
- renomme `Database::getTransactionIdsContainingClusterInSet` en `Database::getFilteredTransactionIdsIfHaveCluster`
- renomme `PatternGenerator::runLcm` en `PatternGenerator::run`

Carmona-Anthony committed:
Ajout d'une classe resultat pour la generation de clusters et d'une
gestion des configs

stardisblue committed:
:fire: debug de generateClusters

Carmona-Anthony committed:
Changement sur closedSwarm detector et creation de database 2

Carmona-Anthony committed:
Merge remote-tracking branch 'origin/feature/solver-manager' into feature/solver-manager

# Conflicts:
#	src/fr/jgetmove/jgetmove/solver/ClusterGenerator.java

stardisblue committed:
ptite correction du siècle

Carmona-Anthony committed:
prepare l'arrivée de patterngenerator

Carmona-Anthony committed:
Refactor ConvoyDetector et PatternGenerator

Carmona-Anthony committed:
Debug PatternGenerator et Solver en enlevant Database2

stardisblue committed:
:fireworks: micro-debugging

stardisblue committed:
simplification de la detection :)
